python/cudf/cudf/__init__.py

```diff
@@ -101,10 +101,16 @@
 from cudf.core.tools.datetimes import date_range
 from cudf.utils.dtypes import _NA_REP
 from cudf.utils.utils import set_allocator
+from ptxcompiler.patch import patch_numba_codegen_if_needed
 
 cuda.set_memory_manager(rmm.RMMNumbaManager)
 cupy.cuda.set_allocator(rmm.rmm_cupy_allocator)
 
+# Patch Numba to support CUDA enhanced compatibility.
+# See https://github.com/rapidsai/ptxcompiler for
+# details.
+patch_numba_codegen_if_needed()
```

Comment:
Does this need `del`ing so users don't see a `cudf.patch_numba_codegen_if_needed` function?

Reply:
nice catch! Fixed.

Reply:
Should we be doing this before configuring RMM?

Reply:
Any reason for preferring that order? (I'm not opposed by any means)

Reply:
Resolving for now; we can revisit if this order breaks us in unexpected ways.  @gmarkall mentioned that _when_ the runtime patching is done shouldn't matter so much.